### PR TITLE
Removes depracated Contract.eventFilter. Fixes #1028.

### LINKED ISSUE #1028: Remove `contract.eventFilter`
* Version: 4.6

### What was wrong?

contract.eventFilter was deprecated in v4, and can be removed in v5.

### PR DESCRIPTION
**Pull request by Exef**
### What was wrong?

Contract.eventFilter() method marked as deprecated and it should be removed in v5.
Related to Issue #1028 

### How was it fixed?

I removed the deprecated function, the test that used it and documentation about it.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by Exef** [comment]
Thank you, @dylanjw for fixing errors. 

Can you tell me, what have I done wrong with test fixture?

**Comment by dylanjw** [comment]
Thanks for contributing!

pytest fixtures can be parametrized, so that everytime a test uses that fixture, it is run with each parameter.

The fixture decorator `@pytest.fixture(params=['v3', 'v4'])` was passing both 'v3' and 'v4' values to `return_filter_by_api`.  Because the 'v3' conditional was removed, in the case when 'v3' was passed the ValueError on this line https://github.com/ethereum/web3.py/pull/1032/commits/523f4938317282a60ea71688ec50ad400304a7cd#diff-e3a6abe3e92df49f9a94069e9a05f9e7L147 was being raised.

**Comment by pipermerriam** [comment]
@dylanjw before this is merged we'll need to get a legacy `v4` branch created to allow merging of these new `v5` only features into `master` (and then anything that is a bugfix needs to be backported to `v4`)

**Comment by dylanjw** [comment]
The https://github.com/ethereum/web3.py/tree/v4 branch is up to date with 4.7.1

File: docs/contracts.rst [hunk 1]
```diff
@@ -484,42 +439,6 @@ and the arguments are ambiguous.
         1
 
 
-
-.. _event-log-object:
```
Comment (veox) [inline docs/contracts.rst]: Slug is referenced elsewhere in the docs (via `:ref:`), which produces a warning.

Section below also referenced.

**Pull request merged**